lib/devices/Viomi.js

```diff
@@ -90,15 +88,23 @@ class Viomi extends MiioVacuum {
 
     pollMap() {
         clearTimeout(this.pollMapTask);
-        this.sendLocalWithPromise("set_uploadmap", [2], {timeout: 2000}).finally(() => {
+        this.sendLocal("set_uploadmap", [2], {timeout: 2000}).finally(() => {
             let repollSeconds =
                 MiioVacuum.FAST_MAP_PULLING_STATES.indexOf(this.status.state) !== -1 ? 5 : 300;
             this.pollMapTask = setTimeout(() => this.pollMap(), repollSeconds * 1000);
         });
     }
 
-    preprocessMap(data, callback) {
-        return zlib.inflate(data, callback);
+    preprocessMap(data) {
+        return new Promise((resolve, reject) => {
+            zlib.inflate(data, (err, result) => {
+                if(err) {
+                    reject(err);
+                }else{
```

Comment:
(nit-picking) there should be spaces here after `if` and around `else`
could also write this less verbosely with `err ? reject(err) : resolve(result)`

Reply:
fixed and added eslint keyword-spacing rule